website/docs/tutorials/07-testing/03-fuzzing/README.md

```diff
@@ -0,0 +1,248 @@
+# Fuzzing
+
+:::warning
+This feature is actively developed and many new additions will land in future Protostar releases.
+:::
+
+Protostar tests can take parameters, which makes such tests to be run in a _fuzzing mode_.
+In this mode, Protostar treats the test case parameters as a specification of the test case,
+in the form of properties which it should satisfy,
+and tests that these properties hold in numerous randomly generated input data.
+
+This technique is often called _property-based testing_.
+From the perspective of a user, the purpose of property-based testing is to make it easier for the
+user to write better tests.
+
+## Example
+
+### The _Safe_
+
+Let's see how fuzzing works in Protostar, by writing a test for an abstract "safe":
+
+```cairo title="src/main.cairo"
+%lang starknet
+from starkware.cairo.common.math import assert_nn
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+@storage_var
+func balance() -> (res : felt):
+end
+
+@external
+func withdraw{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr
+}(amount : felt):
+    if amount == 0:
+        return ()
+    end
+
+    let (res) = balance.read()
+    let new_res = res - amount
+
+    with_attr error_message("Cannot withdraw more than stored."):
+        assert_nn(new_res)
+    end
+
+    balance.write(new_res)
+    return ()
+end
+
+@constructor
+func constructor{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr
+}():
+    balance.write(10000)
+    return ()
+end
+```
+
+### Unit testing
+
+Let's first verify this function by writing a unit test in order to find the general property we are
+testing for:
+
+```cairo title="tests/test_main.cairo"
+%lang starknet
+from src.main import balance, withdraw
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+
+@external
+func test_withdraw{
+    syscall_ptr : felt*,
+    pedersen_ptr : HashBuiltin*,
+    range_check_ptr
+}():
+    alloc_locals
+    let (pre_balance_ref) = balance.read()
+    local pre_balance = pre_balance_ref
+
+    let amount = 1
+
+    withdraw(amount)
+
+    let (post_balance) = balance.read()
+    assert post_balance = pre_balance - amount
+
+    return ()
+end
+```
+
+So far, so good. Running the test, we see it passes:
+
+```text title="$ protostar test"
+[PASS] tests/test_main.cairo test_withdraw (steps=129)
+       range_check_builtin=1
+
+11:28:43 [INFO] Test suites: 1 passed, 1 total
+11:28:43 [INFO] Tests:       1 passed, 1 total
+```
+
+### Generalizing the test
+
+This unit test does test that we can withdraw "some" amount from our safe.
+However, can we be sure that it works for all amounts, not just 1?
```

Comment:
```suggestion
However, can we be sure that it works for all amounts, not just this particular one?
```
it may suggests you have `1` as number in mind